sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -115,13 +115,15 @@ class
   device_global &operator=(const device_global &) = delete;
   device_global &operator=(const device_global &&) = delete;
 
-  multi_ptr<T, access::address_space::global_space> get_multi_ptr() noexcept {
+  multi_ptr<T, access::address_space::global_space, access::decorated::legacy>
```

Comment:
To avoid breaking existing user code the new decoration template argument defaults to `legacy`. This both ensures that their templates resolve and that the interface stays as they expect until we are ready to remove it.

You are right that for this case we should make the transition to decorations directly, but for other extensions (like [sycl_ext_oneapi_local_memory](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc)) where the their `multi_ptr` interfaces don't mention it, we need to at least default to legacy. We could add the template argument and default it, but I would rather that is something we do when it is decided in the extension.

I will fix it for `device_global` though! Thanks for pointing it out!